Sources/Utility/ArgumentParser.swift

```diff
@@ -382,12 +361,94 @@ public final class ArgumentParser {
         }
     }
 
+    /// An enum representing way to parse argument values.
+    public enum ParsingStrategy {
+        case one
+        case multiple
+        case remainder
+        
+        func parse(_ parser: inout Parser, kind: ArgumentKind.Type) throws -> [ArgumentKind] {
+            var values: [ArgumentKind] = []
+            
+            switch self {
+            case .one:
+                var parserProtocol = parser as ArgumentParserProtocol
+                try values.append(kind.init(parser: &parserProtocol))
```

Comment:
Is it a compiler error to pass `&parser` directly? The force cast below seems unfortunate